{{cookiecutter.project_slug}}/pyproject.toml

```diff
@@ -0,0 +1,234 @@
+{%- set license_classifiers = {
+    'MIT license': 'License :: OSI Approved :: MIT License',
+    'BSD license': 'License :: OSI Approved :: BSD License',
+    'ISC license': 'License :: OSI Approved :: ISC License (ISCL)',
+    'Apache Software License 2.0': 'License :: OSI Approved :: Apache Software License',
+    'GNU General Public License v3': 'License :: OSI Approved :: GNU General Public License v3 (GPLv3)'
+} -%}
+[build-system]
+requires = ["flit_core >=3.8,<4"]
+build-backend = "flit_core.buildapi"
+
+[project]
+name = "{{ cookiecutter.project_slug }}"
+authors = [
+  {name = "{{ cookiecutter.full_name.replace('\"', '\\\"') }}", email = "{{ cookiecutter.email }}"}
+]
+maintainers = []
+readme = {file = "README.rst", content-type = "text/x-rst"}
+requires-python = ">=3.8.0"
+keywords = ["{{ cookiecutter.project_slug }}"]
+{%- if cookiecutter.open_source_license in license_classifiers %}
+license = {file = "LICENSE"}
+{%- endif %}
+classifiers = [
+  "Development Status :: 2 - Pre-Alpha",
+  "Intended Audience :: Developers",
+{%- if cookiecutter.open_source_license in license_classifiers %}
+  "{{ license_classifiers[cookiecutter.open_source_license] }}",
+{%- endif %}
+  "Natural Language :: English",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11"
+]
+dynamic = ["description", "version"]
+dependencies = [
+  {% if cookiecutter.command_line_interface|lower == 'click' %}"Click>=8.0"{%- endif %}
+]
+
+[project.optional-dependencies]
+dev = [
+  # Dev tools and testing
+  "pip>=23.1.2",
+  "bump2version>=1.0.1",
+  "watchdog>=3.0.0",
+  "flake8>=6.1.0",
+  "flake8-rst-docstrings>=0.3.0",
+  "flit",
+  "tox>=4.5.1",
+  "coverage>=7.2.5",
+  "coveralls>=3.3.1",
+  "sphinx>=7.0.1",
+  {% if cookiecutter.command_line_interface|lower == 'click' -%}
+  "click>=8.1.7",
+  {%- endif %}
+  {% if cookiecutter.use_pytest == 'y' -%}
+  "pytest>=7.3.1",
+  "pytest-cov>=4.0.0",
+  {%- endif %}
+  {% if cookiecutter.use_black == 'y' -%}
+  "black>=23.10.1",
+  "isort>=5.12.0",
+  {%- endif %}
+  "pre-commit>=3.3.2"
+]
+docs = [
```

Comment:
Since we have this here, do we still need `environment-docs.yml`? You left it untouched.

Reply:
I don't think we lose anything by having both the `pyproject.toml` and `environment-docs.yml`. It's convenient to be able to install the docs without conda.